objects/coffee_mug.rb

```diff
@@ -0,0 +1,34 @@
+# This is a coffee mug class
+class Coffee_Mug
+  def initialize
+    @cup_is_full = false
+    @cup_is_hot = false
+
+  end
+
+  def get_coffee
+    if @cup_is_full
+      puts "Mug is full"
+    else
+      @cup_is_full = true
+      @cup_is_hot = true
+    end
+
+  end
+
+  def is_coffee_hot
+    puts @cup_is_hot
+  end
+end
+
+
+# Implementation of the coffee mug class
+mug1 = Coffee_Mug.new
```

Comment:
mug1 = CoffeeMug.new
    mug1.fill     ( This could fill the cup with 12 oz of coffee, and set the time the coffee was filled)
    mug1.sip    (This should remove one oz of coffee, if the coffee is warm/hot)
    mug1.sip
    mug1.sip    (Coffee is now warm because it's been greater than 5 minutes)
    mug1.refill  (Reset oz to 12, time filled to now)

Have some fun with it. :)